### ISSUE TITLE
Stopped working today 6.20

### ISSUE THREAD

**Issue by imbader** [iamfugui/youtube-adb#24]

**Comment by WiLLiWOficial** [comment]
It stopped working. Temporarily, you must press F5 three, four or five times to release the message "you are using an ad blocker..." and watch the video.

動作しなくなりました。一時的に、F5 を 3、4、5 回押して、「広告ブロッカーを使用しています...」というメッセージを解除し、ビデオを視聴する必要があります。

**Comment by Papqwa** [comment]
No longer working. Need to reload page for video to load and also getting  the message "you are using an ad blocker..."

**Comment by iambumblehead** [comment]
this is a copy of the ad blocker popup markup

<details>
  <summary>ad blocker popup</summary>

```html
<ytd-popup-container class="style-scope ytd-app">
  <!--css-build:shady-->
  <!--css-build:shady-->
  <iron-media-query class="style-scope ytd-popup-container" style="display: none;">
  </iron-media-query>
  <tp-yt-paper-dialog style-target="host" role="dialog" tabindex="-1" class="style-scope ytd-popup-container" prevent-autonav="true" modern="" style="outline: none; z-index: 2202; position: fixed; top: 5.25195px; left: 0.00390625px; box-sizing: border-box; max-height: 551.006px; max-width: 467px;">
    <!--css-build:shady-->
    <ytd-enforcement-message-view-model class="style-scope ytd-popup-container" tabindex="-1" style="outline: none;">
      <!--css-build:shady--><!--css-build:shady-->
      <div id="container" class="style-scope ytd-enforcement-message-view-model">
        <div id="header" class="style-scope ytd-enforcement-message-view-model">
          <yt-image preserve-height="" class="style-scope ytd-enforcement-message-view-model" height="31" id="logo-icon-with-padding">
            <img alt="" height="31" class="yt-core-image yt-core-image--fill-parent-width yt-core-image--content-mode-scale-aspect-fill yt-core-image--loaded" src="https://www.gstatic.com/youtube/img/promos/growth/e40b77821226f0d031438cc0a8a6fdb5908d59f63626c6c06d5d965f89a72255_218x62.png">
          </yt-image>
          <yt-button-view-model></yt-button-view-model>
        </div>

        <div id="header-image" class="style-scope ytd-enforcement-message-view-model" hidden="">
          <yt-image height="64" preserve-height="" class="style-scope ytd-enforcement-message-view-model">
          </yt-image>
        </div>

        <div id="title" class="style-scope ytd-enforcement-message-view-model">
          <yt-attributed-string class="style-scope ytd-enforcement-message-view-model">
            <span class="yt-core-attributed-string yt-core-attributed-string--white-space-pre-wrap" role="text">
              <span class="" style="">広告ブロッカーは YouTube の利用規約に違反しています</span>
            </span>
          </yt-attributed-string>
        </div>

        <ul id="body" class="style-scope ytd-enforcement-message-view-model">
          <li class="style-scope ytd-enforcement-message-view-model">
            <yt-attributed-string class="body-message style-scope ytd-enforcement-message-view-model">
              <span class="yt-core-attributed-string yt-core-attributed-string--white-space-pre-wrap" role="text">
                <span class="" style="">
                  広告ブロッカーを使用されていませんか？ YouTube を許可リストに登録する場合や、広告ブロッカーを無効する場合を除き、動画の再生がブロックされます。
                </span>
              </span>
            </yt-attributed-string>
          </li>
          
          <li class="style-scope ytd-enforcement-message-view-model">
            <yt-attributed-string class="body-message style-scope ytd-enforcement-message-view-model">
              <span class="yt-core-attributed-string yt-core-attributed-string--white-space-pre-wrap" role="text">
                <span class="" style="">
                  世界で数十億人に上るユーザーが YouTube を使えるのは、広告のおかげです。
                </span>
              </span>
            </yt-attributed-string>
          </li>
          
          <li class="style-scope ytd-enforcement-message-view-model">
            <yt-attributed-string class="body-message style-scope ytd-enforcement-message-view-model">
              <span class="yt-core-attributed-string yt-core-attributed-string--white-space-pre-wrap" role="text">
                <span class="" style="">
                  YouTube Premium に加入すると、広告なしでもクリエイターがあなたの視聴によって収益を得ることをサポートできます
                </span>
              </span>
            </yt-attributed-string>
          </li>
          <ps-dom-repeat as="item" class="style-scope ytd-enforcement-message-view-model">
            <template is="dom-repeat" class="style-scope ytd-popup-container">
            </template>
          </ps-dom-repeat>
        </ul>

        <div id="buttons" class="style-scope ytd-enforcement-message-view-model">
          <yt-button-view-model class="ytd-enforcement-message-view-model action-button">
            <button-view-model class="yt-spec-button-view-model">
              <button class="yt-spec-button-shape-next yt-spec-button-shape-next--filled yt-spec-button-shape-next--overlay yt-spec-button-shape-next--size-m yt-spec-button-shape-next--enable-backdrop-filter-experiment" aria-disabled="false" aria-label="YouTube 広告をブロックするのをやめる" title="" style="">
                <div class="yt-spec-button-shape-next__button-text-content">
                  YouTube 広告をブロックするのをやめる
                </div>
                <yt-touch-feedback-shape style="border-radius: inherit;">
                  <div class="yt-spec-touch-feedback-shape yt-spec-touch-feedback-shape--overlay-touch-response-inverse" aria-hidden="true">
                    <div class="yt-spec-touch-feedback-shape__stroke" style="">
                    </div>
                    <div class="yt-spec-touch-feedback-shape__fill" style="">
                    </div>
                  </div>
                </yt-touch-feedback-shape>
              </button>
            </button-view-model>
          </yt-button-view-model>
          <yt-button-view-model class="ytd-enforcement-message-view-model action-button" dialog-dismiss="">
            <button-view-model class="yt-spec-button-view-model">
              <a class="yt-spec-button-shape-next yt-spec-button-shape-next--tonal yt-spec-button-shape-next--overlay yt-spec-button-shape-next--size-m yt-spec-button-shape-next--enable-backdrop-filter-experiment" aria-disabled="false" aria-label="YouTube Premium を試す" title="" href="/premium" target="" force-new-state="true" style="">
                <div class="yt-spec-button-shape-next__button-text-content">
                  YouTube Premium を試す
                </div>
                <yt-touch-feedback-shape style="border-radius: inherit;">
                  <div class="yt-spec-touch-feedback-shape yt-spec-touch-feedback-shape--overlay-touch-response" aria-hidden="true">
                    <div class="yt-spec-touch-feedback-shape__stroke" style="">
                    </div>
                    <div class="yt-spec-touch-feedback-shape__fill" style="">
                    </div>
                  </div>
                </yt-touch-feedback-shape>
              </a>
            </button-view-model>
          </yt-button-view-model>
        </div>

        <div id="divider" class="style-scope ytd-enforcement-message-view-model"></div>

        <div id="feedback" class="style-scope ytd-enforcement-message-view-model">
          <yt-attributed-string id="feedback-message" class="style-scope ytd-enforcement-message-view-model">
            <span class="yt-core-attributed-string yt-core-attributed-string--white-space-pre-wrap">
              <span class="" style="">
                広告ブロッカーを使用していない場合:
              </span>
              <span class="yt-core-attributed-string--link-inherit-color" style="color: rgb(62, 166, 255);">
                <a class="yt-core-attributed-string__link yt-core-attributed-string__link--call-to-action-color" tabindex="0" role="button" href="javascript:void(0);">
                  問題を報告
                </a>
              </span>
            </span>
          </yt-attributed-string>
        </div>
      </div>
    </ytd-enforcement-message-view-model>
  </tp-yt-paper-dialog>
</ytd-popup-container>
```
</details>

maybe a separate script altogether should be used to block youtube popups.

I think if I were going to try and resolve this I would script it to remove any element that is a direct parent of `.ytd-popup-container .ytd-enforcement-message-view-model`

**Comment by iambumblehead** [comment]
this works here. its fragile, but, works...
```javascript
    let resumeVideo = () => {
      const videoelem = document.body.querySelector('video.html5-main-video')
      if (videoelem && videoelem.paused) {
         console.log('resume video')
         videoelem.play()
      }
    }

    let removePop = node => {
       const elpopup = node.querySelector('.ytd-popup-container > .ytd-popup-container > .ytd-enforcement-message-view-model')
   
       if (elpopup) {
         elpopup.parentNode.remove()
         console.log('remove popup', elpopup)
         const bdelems = document
           .getElementsByTagName('tp-yt-iron-overlay-backdrop')
         for (var x = (bdelems || []).length; x--;)
           bdelems[x].remove()
         resumeVideo()
       }

       if (node.tagName.toLowerCase() === 'tp-yt-iron-overlay-backdrop') {
         node.remove()
         resumeVideo()
         console.log('remove backdrop', node)
       }
    }
    let obs = new MutationObserver(mutations => mutations.forEach(mutation => {
      if (mutation.type === 'childList') {
        Array.from(mutation.addedNodes)
          .filter(node => node.nodeType === 1)
          .map(node => removePop(node))
      }
    }))

    // have the observer observe foo for changes in children
    obs.observe(document.body, {
      childList: true,
      subtree: true
    })
```
